src/components/App.js

```diff
@@ -1,14 +1,154 @@
 import React from 'react';
 import { hot } from 'react-hot-loader/root';
-import macaroon from '../assets/img/Macaroonicon.png';
-
-function App() {
-  return (
-    <div className="app">
-      <h1 className="title">Shmotify.</h1>
-      <img src={macaroon} alt="" />
-    </div>
-  );
+import { checkURIforError, getTokenFromURI } from '../config/authConfig';
+import { Dashboard } from './Dashboard';
+import { UserPlaylists } from './UserPlaylists';
+import { Sidebar } from './Sidebar';
+import { Welcome } from './Welcome';
+import { LoginLink } from './LoginLink';
+
+
+class App extends React.Component {
+  state = {
+    token: '',
+    user: {
+      name: '',
+      error: false,
+      errorMessage: '',
+    },
+    playlists: {
+      items: [],
+      nextPlaylistsEndpoint: null,
+      error: false,
+      errorMessage: '',
+    },
+  }
+
+  componentDidMount() {
+    const authError = checkURIforError();
+    const token = getTokenFromURI();
+    this.setAuthError(authError);
+    if (token) {
+      this.setToken(token);
+      this.getProfile(token);
+      this.getPlaylists(token);
+    }
+  }
+
+  componentDidUpdate() {
+    const { token, playlists: { nextPlaylistsEndpoint } } = this.state;
+    if (nextPlaylistsEndpoint) {
+      this.getPlaylists(token, nextPlaylistsEndpoint);
+    }
+  }
+
+  setToken = (token) => {
+    this.setState({ token });
+  }
+
+  setAuthError = (error) => {
+    const errorMessage = error
+      ? 'Profile access revoked by user.'
+      : '';
+    this.setState(prevState => ({
+      ...prevState,
+      user: {
+        error,
+        errorMessage,
+      },
+    }));
+  }
+
+  getProfile = (token) => {
+    const myHeaders = new Headers();
+    myHeaders.append('Authorization', `Bearer ${token}`);
+
+    fetch('https://api.spotify.com/v1/me', {
+      method: 'GET',
+      headers: myHeaders,
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        throw Error(`Request rejected with status ${response.status}`);
+      })
+      .then((userData) => {
+        const name = userData.display_name;
+        this.setState(prevState => ({
+          ...prevState,
+          user: {
+            name,
+            error: false,
+            errorMessage: '',
+          },
+        }));
+      })
+      .catch((error) => {
+        this.setState({
+          user: {
+            error: true,
+            errorMessage: error.message,
+          },
+        });
+      });
+  }
+
+  getPlaylists = (token, endpoint = 'https://api.spotify.com/v1/me/playlists') => {

```

Comment:
It feels a little smelly writing a separate method for each different API call, but a better way has not jumped out at me yet. I am open to guidance on this.